lib/msal-browser/src/interaction_client/StandardInteractionClient.ts

```diff
@@ -173,22 +173,44 @@ export abstract class StandardInteractionClient extends BaseInteractionClient {
      * @param requestAuthority
      * @param requestCorrelationId
      */
-    protected async getDiscoveredAuthority(requestAuthority?: string): Promise<Authority> {
+    protected async getDiscoveredAuthority(requestAuthority?: string, requestAzureAuthOptions?: AzureAuthOptions): Promise<Authority> {
         this.logger.verbose("getDiscoveredAuthority called");
         const authorityOptions: AuthorityOptions = {
             protocolMode: this.config.auth.protocolMode,
             knownAuthorities: this.config.auth.knownAuthorities,
             cloudDiscoveryMetadata: this.config.auth.cloudDiscoveryMetadata,
-            authorityMetadata: this.config.auth.authorityMetadata
+            authorityMetadata: this.config.auth.authorityMetadata,
         };
 
-        if (requestAuthority) {
+        // build authority string based on auth params, precedence - azureCloudInstance + tenant >> authority and request parameters  >> config parameters
+        let userRequestedAuthority;
+        let authorityAzureCloudInstance;
+
+        // Fetch the authority from request if provided
+        if (requestAuthority || requestAzureAuthOptions) {
```

Comment:
All this logic should live in the Authority class imo

Reply:
Moved